soci/soci_index.go

```diff
@@ -182,19 +218,40 @@ func BuildSociIndex(ctx context.Context, cs content.Store, img images.Image, spa
 		IndexAnnotationBuildToolVersion:    config.buildToolVersion,
 	}
 
-	sociIndex := SociIndex{
-		MediaType:    sociIndexMediaType,
+	refers := &ocispec.Descriptor{
+		MediaType:   imgManifestDesc.MediaType,
+		Digest:      imgManifestDesc.Digest,
+		Size:        imgManifestDesc.Size,
+		Annotations: imgManifestDesc.Annotations,
+	}
+	return NewIndex(ztocsDesc, refers, annotations, config.manifestType), nil
+}
+
+// Returns a new index.
+func NewIndex(blobs []ocispec.Descriptor, subject *ocispec.Descriptor, annotations map[string]string, manifestType ManifestType) *Index {
```

Comment:
Maybe break this into two different functions and have `NewIndex` switch between them based on `config.Type?`? `NewIndexForOCI` and `NewIndexForORAS` while it dupes logic a bit it should simplify and keep some code together (easier to rip out or extend then an if statement)

Reply:
Sure.